### DISCUSSION TITLE
wifi connectivity lost when using app layer

### DISCUSSION THREAD

**Discussion by jmiguelsangar** [bitcraze/discussions#335] [Q&A]
Hi everyone, 
I am trying to do the following:
Send a command from STM32 to GAP 8 via CPX, take a photo and send it via wifi to a host.

I'm using the code from the wifi-streaming example that works perfectly as a guidance, but I just want to send a single photo every time and record it in the host. For the STM part I'm using the app layer as recomended. At this time, just to test if it the flash is done correctly, I just build and flash a simple code into STM using app layer (the appMain method really does nothing for the moment). The problem that I've come across is that when I flash it, the wifi connection dissapears, and I have to reload the crazyflie firmware to make wifi work again and connect the host to the AIdeck. The app layer is supposed to be out of the main firmware and it shouldn't interfere. Any idea about this?

Do you think feasible sending a command using CPX in a two way communication between STM to GAP8, or do you suggest another way of doing this?. I have not found simple examples on how to use this protocol to send data between MCUs. After studying another ideas I wanted to try this protocol.

Thanks in advance for your great help in this forum. It really helps to understand and step by step keep on going on the projects.

**Comment by knmcguire** [comment]
Hi! 

Could you let us know the exact process how you make the app? Perhaps the kbuild configuration is changed for the aideck's WiFi credentials.

**Reply by jmiguelsangar** [reply]
![2022-10-25_19h32_38](https://user-images.githubusercontent.com/7648632/197843991-7d15aa3f-2db0-4ef4-9b6a-1efe19826926.png)
![2022-10-25_19h33_03](https://user-images.githubusercontent.com/7648632/197843996-daa74048-6499-44a6-b55d-7d52f4ac513e.png)
![2022-10-25_19h33_13](https://user-images.githubusercontent.com/7648632/197843998-fc0c1216-8ff1-4441-abea-f1d689d95b7c.png)
![2022-10-25_19h33_24](https://user-images.githubusercontent.com/7648632/197844001-393fafb9-0d13-479a-95e7-683ea7110aea.png)
![2022-10-25_19h33_37](https://user-images.githubusercontent.com/7648632/197844002-d7ac65b5-1260-430c-963b-fea36b8648f1.png)

I send you some screenshots. First I load the firmware and the wifi-img-streaming example. I use the crazyflie as an acces point and I can connect and make the example work without any problem. At this point if I load the dron-navigation code the Accespoint in my host wifi conections dissapear. My challenge is to send signals from the crazyflie to the AIdeck to take a photo and send it to the host.

Thank you

**Reply by krichardsson** [reply]
Try to add a `vTaskDelay(M2T(1000));` in your while loop.
In the current implementation the while loop is depleting the system of resources.

**Reply by krichardsson** [reply]
Tip of today: using screen dumps makes it impossible to copy/paste your code when answering, a better option is to use code blocks instead. Just use three back ticks and an optional language identifier, like this:

    ```c
    int myNiceCFunction(int aVariable) {
      return 4711;
    }
    ```
 

will look like this

```c
int myNiceCFunction(int aVariable) {
  return 4711;
}
```

**Reply by jmiguelsangar** [reply]
Hi, thank you for your help. I have tried adding the sentence inside the while loop but the access point dissapears and I cannot connect the host to the crazyflie.

**Reply by evoggy** [reply]
Hi! Could you check that you're not setting up the wifi from the GAP8 as well? I know there's an issue when you try to set up the wifi a second time.

**Reply by jmiguelsangar** [reply]
Let me check again and I'll let you know asap
Thank you

**Reply by jmiguelsangar** [reply]
Hi, I´ve been testing againg and it is a bit strange. Crazyflie access point dissappears when I load my code using APP Layer. I have just put a debug line in my code to test it in the cf console and it works when the drone is started, but access point dissapears. If a reload the firmware into SMT32 the access point appears again. I cannot find what´s going on when I load my code using App layer. 

Thank you

**Reply by evoggy** [reply]
The only thing I can think of is that the access point is being set up twice, this can happen if both the GAP8 (with the wifi-streaming-example binary) sets up the access point and then the Crazyflie also does this (set up via menuconfig). Can you try to switch off setting up the access point in the Crazyflie firmware (via menuconfig) to see if this helps?
